### ISSUE TITLE
Alternate implementation for pick_action

### ISSUE THREAD

**Issue by tbnorth** [tscv11/leo4sqlite#11]
Nothing really wrong with the current, but given the uniform structure of self.<method>(c), I'd probably do something like:
```
actions = {
    "import table": "get_ext_db select_table get_blob_col get_layout make_nodes grand_central",
    "view blob": "get_ext_db select_table get_blob_col view_blob",
    # add others here
}

if c._leo4sqlite['action'] not in actions:
    raise UnknownActionError()
for step in actions[c._leo4sqlite['action'].split()]:
    method = getattr(self, step)
    method(c)

**Comment by tscv11** [comment]
I will study this, thank you.

**Comment by tscv11** [comment]
Okay...that was fun! I added a little twist. What do you think?

**Comment by tbnorth** [comment]
Yep, the `cmds` thing is cool, exactly the sort of "higher level language" thing that can evolve when you take that route.  I don't think any of the `()`s in the `cmds =` and `actions =` are needed.

Another neat trick, when you make a commit closing an issue like this, you can just say
```
fixed #11, blah blah
```
in the commit msg. and GitHub will automagically notice that, close the issue, and link to it from the issue page, e.g. https://github.com/leo-editor/leo-editor/issues/591 where it says "tbnorth closed this in e29de68 on Nov 27, 2017".
